build-tools/install-additional-software.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/sh
+
+#
+# Copyright 2019-2020 JetBrains s.r.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Uncomment this in case uou need extra debug info
+#set -e # Any command which returns non-zero exit code will cause this shell script to exit immediately
+#set -x # Activate debugging to show execution details: all commands will be printed before execution
+
+DOWNLOAD_URL=$1
+# Define filter function
+function string_filter {
+  sed -z 's/  */ /g;s/\n//g' $1
+}
+
+# In SOFTWARE_LIST block add software package names. This string was directly passed to `apt install` command
+# In ADDITIONAL_COMMANDS you can add any BASH commands. Please, note that you need to escape $ character by backslash (\)
+
+# Detect type of IDE we use and create a list of required software
+case $DOWNLOAD_URL in
+
+#
+# Block for CLion
+  *CLion*)
+    SOFTWARE_LIST=$(string_filter <<- EOM
+      build-essential
+      clang
+EOM
+)
+    ADDITIONAL_COMMANDS=$(string_filter <<- EOM
+
+EOM
+    );;
+
+#
+# Block for GoLand
+  *goland*)
+    SOFTWARE_LIST=$(string_filter <<- EOM
+
+EOM
+)
+    ADDITIONAL_COMMANDS=$(string_filter <<- EOM
+
+EOM
+    );;
+
+#
+# Block for DataGrip
+  *datagrip*)
+    SOFTWARE_LIST=$(string_filter <<- EOM
+
+EOM
+)
+    ADDITIONAL_COMMANDS=$(string_filter <<- EOM
+
+EOM
+    );;
+
+#
+# Block for PhpStorm
+  *PhpStorm*)
+    SOFTWARE_LIST=$(string_filter <<- EOM
+
+EOM
+)
+    ADDITIONAL_COMMANDS=$(string_filter <<- EOM
+
+EOM
+    );;
```

Comment:
A lot of duplication, and I have no idea how to make it more compact. If you have, please do

Reply:
I have some ideas about this. In few days I`ll try to add smth

Reply:
@Avvessalom unfortunately I cant optimize this(

For now I have only one solution without code duplcation, but it required a many (at least one for each variant of IDE) files, witch store software list for each IDE.
But I think this is not a good solution. 

Maybe we can accept this solution and, in future, give more convinient way?

Reply:
Discussed here: https://github.com/JetBrains/projector-docker/pull/16#discussion_r754174125. So let's leave it if we can't improve it more.